### PR TITLE
Fix and re-enable XSLT checksum tests

### PR DESCRIPTION
**Pull request by sepidehMS**
Fixes https://github.com/dotnet/corefx/issues/9877

Many XSLT tests used to calculate checksums to verify the result of a transformation. This type of verification caused flakiness in these tests. In this PR, we replace that verification (comparison with an expected checksum value) by comparing the transformed xml with the expected xml using `XmlDiff`.

cc: @danmosemsft @stephentoub @AlexGhiondea @joperezr

### CODE REVIEW

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs [hunk 1]
```diff
@@ -586,31 +586,40 @@ public int LoadXSL_Resolver(String _strXslFile, XmlResolver xr)
...
+                }
+                
+                //Output the expected and actual values
+                _output.WriteLine("Expected : " + expectedValue);
+                _output.WriteLine("Actual : " + actualValue);
+                
+                bool bResult;
+
+                //Load into XmlTextReaders
+                using (XmlTextReader tr1 = new XmlTextReader("out.xml"))
```
Comment (danmosemsft) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: >"out.xml" [](start = 61, length = 9)

_strOutFile

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs [hunk 2]
```diff
@@ -586,31 +586,40 @@ public int LoadXSL_Resolver(String _strXslFile, XmlResolver xr)
...
 
-            if (bResult)
-                return;
-            else
-                Assert.True(false);
+                if (bResult)
+                    return;
+                else
+                    Assert.True(false);
+            }
```
Comment (danmosemsft) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: replace with 
Assert.True(bResult)

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs [hunk 3]
```diff
@@ -586,31 +586,40 @@ public int LoadXSL_Resolver(String _strXslFile, XmlResolver xr)
...
+            lock(s_outFileMemoryLock)
+            {
+                XmlDiff.XmlDiff xmldiff = new XmlDiff.XmlDiff();
+                xmldiff.Option = XmlDiffOption.InfosetComparison | XmlDiffOption.IgnoreEmptyElement | XmlDiffOption.NormalizeNewline;
+                
+                string actualValue;
+                
+                using (FileStream outFile = new FileStream("out.xml", FileMode.Open, FileAccess.Read))
+                {
+                    StreamReader sr = new StreamReader(outFile);
```
Comment (danmosemsft) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: or replace all with File.ReadAllText

---
In reply to: [90154953](https://github.com/dotnet/corefx/pull/14113#discussion_r90154953) [](ancestors = 90154953)

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs [hunk 4]
```diff
@@ -586,31 +586,40 @@ public int LoadXSL_Resolver(String _strXslFile, XmlResolver xr)
...
+                    actualValue = sr.ReadToEnd();
+                    sr.Dispose();
+                }
+                
+                //Output the expected and actual values
+                _output.WriteLine("Expected : " + expectedValue);
+                _output.WriteLine("Actual : " + actualValue);
+                
+                bool bResult;
```
Comment (danmosemsft) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: >bResult [](start = 21, length = 7)

no new uses of hungarian prefixes

**Comment by danmosemsft** [comment]
>         }

dead now right?

---
Refers to: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs:988 in bb745f1. [](commit_id = bb745f1f6e529c40210924924fef7916a4167b99, deletion_comment = False)

**Comment by danmosemsft** [comment]
:shipit:

**Comment by danmosemsft** [comment]
Thanks for doing this @sepidehMS what a pain.

File: src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/XslCompiledTransform.cs [hunk 5]
```diff
@@ -1108,18 +1108,23 @@ public void XmlResolver6()
         */
 
         //[Variation(id = 7, Desc = "document() has absolute URI", Pri = 0)]
-        [ActiveIssue(9877)]
+        [ActiveIssue(14071)]
```
Comment (joperezr) [inline src/System.Private.Xml/tests/Xslt/XslCompiledTransformApi/XslCompiledTransform.cs]: So then this test is still failing?

**Comment by joperezr** [comment]
Just as a general question, did you try running these tests for net46? that might not work due to a different problem (binding redirect) but it's worth the try.

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs (hunk 3 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: Will do.

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs (hunk 1 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: Will do.

Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: Will change.

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs (hunk 2 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: Will do.

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs (hunk 4 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/XSLTransform.cs]: Will change.

File: src/System.Private.Xml/tests/Xslt/XslTransformApi/CXslTransform.cs [hunk 6]
```diff
@@ -2379,15 +2548,21 @@ public void TransformStrStr13()
                 StreamReader fs = null;
 
                 // check if I can open and close the xml file
-                fs = new StreamReader(new FileStream(szFullFilename, FileMode.Open, FileAccess.Read));
-                fs.Dispose();
+                using (FileStream outFile = new FileStream(szFullFilename, FileMode.Open, FileAccess.Read))
+                {
+                    fs = new StreamReader(outFile);
+                    fs.Dispose();
+                }
```
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Xslt/XslTransformApi/CXslTransform.cs]: Will do.

**Comment by sepidehMS** [comment]
@danmosemsft In reply to https://github.com/dotnet/corefx/pull/14113#issuecomment-263795171 there are still a few skipped tests that are using `CheckResult` method. I'm going to clean this up when I resolve those in https://github.com/dotnet/corefx/issues/14071

**Comment by sepidehMS** [comment]
@joperezr In reply to https://github.com/dotnet/corefx/pull/14113#issuecomment-263941132 These tests cannot find runtime assets for net46.

**Comment by sepidehMS** [comment]
Thanks for your feedback. I applied the changes, please take a look.

**Comment by sepidehMS** [comment]
Merging as all tests passed and the only failures was at archiving artifacts step.

**Pull request merged**